app/src/main/java/no/usn/mob3000/ui/screens/user/ProfileScreen.kt

```diff
@@ -35,35 +50,70 @@ import no.usn.mob3000.ui.theme.ProfileUserStatisticsBackground
  * @param onProfileEditClick Callback function to navigate to [ProfileEditScreen].
  * @param onProfileAddFriendsClick Callback function to navigate to [ProfileAddFriendsScreen].
  * @param onProfileFriendRequestsClick Callback function to navigate to [ProfileFriendRequestsScreen].
- * @author frigvid, Hussein
+ * @author frigvid, Husseinabdulameer11
+ * @contributor 258030
  * @created 2024-09-12
  */
 @Composable
 fun ProfileScreen(
-    onProfileEditClick: () -> Unit,
-    onProfileAddFriendsClick: () -> Unit,
-    onProfileFriendRequestsClick: () -> Unit
+    onProfileEditClick: (UserProfile) -> Unit,
+    onProfileAddFriendsClick: (UserProfile) -> Unit,
+    onProfileFriendRequestsClick: () -> Unit,
+    fetchFriends: () -> Unit,
+    fetchUser: (String) -> Unit,
+    fetchUserById: (String) -> Unit,
+    friendState: StateFlow<Result<List<FriendData>>>,
+    userIdState: StateFlow<String?>,
+    userProfilesMap: StateFlow<Map<String, UserProfile>>,
+    setSelectedUser: (UserProfile) -> Unit,
+    authenticationState: StateFlow<AuthenticationState>,
+    authenticationStateUpdate: () -> Unit,
+    onLoginClick: () -> Unit,
+    userGameStats: StateFlow<Result<UserGameStats>>,
+    viewModel: ProfileViewModel
 ) {
+    val friendResult by friendState.collectAsState()
+    val userId by userIdState.collectAsState()
+    val userProfiles by userProfilesMap.collectAsState()
+    val state by remember { authenticationState }.collectAsState()
+    val currentUserProfile by viewModel.currentUserProfile.collectAsState()
+    val gameStats by userGameStats.collectAsState()
+
+    LaunchedEffect(userId) {
+        authenticationStateUpdate()
+        userId?.let { id ->
+            fetchFriends()
+        }
+    }
     Viewport(
         topBarActions = {
-            IconButton(onClick = onProfileEditClick) {
-                Icon(
-                    painter = painterResource(R.drawable.profile_edit),
-                    contentDescription = "Edit Profile",
-                    modifier = Modifier.size(24.dp),
-                    tint = Color.Black
-                )
-            }
+            currentUserProfile?.let { user ->
+                IconButton(onClick = {
+                    setSelectedUser(user)
+                    onProfileEditClick(user)
+                }) {
```

Comment:
Indentation. Multiline parameter lists that have their own body should be line shifted and indented.